apps/framework-cli/src/metrics_inserter.rs

```diff
@@ -0,0 +1,160 @@
+use crate::metrics::MetricEvent;
+use reqwest::Client;
+use serde_json::json;
+use std::sync::Arc;
+use std::time::Duration;
+use tokio::sync::Mutex;
+use tokio::time;
+
+const MAX_FLUSH_INTERVAL_SECONDS: u64 = 10;
+const MAX_BATCH_SIZE: usize = 1000;
+
+pub type BatchEvents = Arc<Mutex<Vec<MetricEvent>>>;
+
+#[derive(Clone)]
+pub struct MetricsInserter {
+    buffer: BatchEvents,
+}
+
+impl MetricsInserter {
+    pub fn new(
+        metric_labels: Option<serde_json::Map<String, serde_json::Value>>,
+        metric_endpoints: Option<serde_json::Map<String, serde_json::Value>>,
+    ) -> Self {
+        let buffer = Arc::new(Mutex::new(Vec::new()));
+
+        tokio::spawn(flush(
+            buffer.clone(),
+            metric_labels.clone(),
+            metric_endpoints.clone(),
+        ));
+
+        Self { buffer }
+    }
+
+    pub async fn insert(&self, event: MetricEvent) -> anyhow::Result<()> {
+        let mut buffer = self.buffer.lock().await;
+        buffer.push(event);
+        Ok(())
+    }
+}
+
+async fn flush(
+    buffer: BatchEvents,
+    metric_labels: Option<serde_json::Map<String, serde_json::Value>>,
+    metric_endpoints: Option<serde_json::Map<String, serde_json::Value>>,
+) {
+    let mut interval = time::interval(Duration::from_secs(MAX_FLUSH_INTERVAL_SECONDS));
+    let client = Client::new();
+
+    loop {
+        interval.tick().await;
+        let mut buffer_owned = buffer.lock().await;
```

Comment:
Not sure if this was what you wanted to do, this will execute every second and will send whatever is in the buffer in chunks.